Sources/HaCWebsiteLib/Models/NewWorkshop+init.swift

```diff
@@ -13,151 +13,154 @@ let filePaths = (
-    contributors = try NewWorkshop.getContributors(from: metadata)
-    thanks = try NewWorkshop.getThanks(from: metadata)
-    furtherReadingLinks = try NewWorkshop.getFurtherReadingLinks(from: metadata)
-    recordingLink = try NewWorkshop.getRecordingLink(from: metadata)
-    slidesLink = try NewWorkshop.getSlidesLink(from: metadata)
-    tags = try NewWorkshop.getTags(from: metadata)
-    license = try NewWorkshop.getLicense(from: metadata)
+    let builder = WorkshopBuilder(localPath: localPath, commitSha: headCommitSha, workshopId: workshopId, metadata: metadata)
+
+    presenterGuide = try builder.getPresenterGuide()
+    promoImageBackground = try builder.getPromoImageBackground()
+    promoImageForeground = try builder.getPromoImageForeground().absoluteString
+    description = try builder.getDescription()
+    prerequisites = try builder.getPrerequisites()
+    setupInstructions = try builder.getSetupInstructions()
+
+    examplesLink = try builder.getExamplesLink()
+    notes = try builder.getNotes()
+
+    title = try builder.getTitle()
+    contributors = try builder.getContributors()
+    thanks = try builder.getThanks()
+    furtherReadingLinks = try builder.getFurtherReadingLinks()
+    recordingLink = try builder.getRecordingLink()
+    slidesLink = try builder.getSlidesLink()
+    tags = try builder.getTags()
+    license = try builder.getLicense()
   }
 
-  fileprivate static func getWorkshopId(localPath: String) throws -> String {
+  private static func getWorkshopId(localPath: String) throws -> String {
     let directoryName = localPath.components(separatedBy: "/").last ?? localPath
 
     if directoryName.hasPrefix("workshop-") {
       return String(directoryName.dropFirst("workshop-".count))
     } else {
-      throw Error.invalidPath
+      throw WorkshopError.invalidPath
     }
   }
 
-  fileprivate static func getMetadata(localPath: String) throws -> Yaml {
+  private static func getMetadata(localPath: String) throws -> Yaml {
     let metadataString: String
     do {
       metadataString = try String(contentsOfFile: localPath + filePaths.metadata, encoding: .utf8)
     } catch {
-      throw Error.missingMetadata
+      throw WorkshopError.missingMetadata
     }
     return try Yaml.load(metadataString)
   }
-
-  fileprivate static func getDescription(localPath: String) throws -> Markdown {
+}
+
+private enum WorkshopError: Swift.Error {
+  case invalidPath
+  case missingMetadata
+  case malformedMetadata(String)
+  case missingDescription
+  case missingPrerequisites
+  case missingPromoImageBackground
+  case missingPromoImageForeground
+  case multiplePromoImageForegrounds
+  case multiplePromoImageBackgrounds
+  case invalidPromoImageForegroundFormat
+  case invalidPromoImageBackgroundFormat
+  case missingNotes
+  case badURLInNotes(String)
+  case emptyExamples
+  case invalidURL(String)
+}
+
+private struct WorkshopBuilder {
+  let localPath: String
+  let commitSha: String?
```

Comment:
When would this be nil?

Reply:
If we ever wanted to do some testing, didn't have a commit sha on hand or something. It's tenuous

Want me to remove it?

Reply:
Can give dud hashes for testing, reckon can make this non-optional